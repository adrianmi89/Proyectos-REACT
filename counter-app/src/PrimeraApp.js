import React from "react";
import PropTypes from "prop-types";//Importando el prototipado

//Añadimos propiedades a la función
const PrimeraApp = ( {persona} ) => {

    const saludo = "Hola Mundo, ";
    const subtitulo = "Zona para los subtítulos";
    const numerosSeparados = [1,2,3,4,5,6];
    const imprimeObjeto = {
        nombre:"Adrian",
        Apellidos:"Martínez Insua",
        Edad:33,
        Estudios:"Desarrollador de Aplicaciones Web"
    }

    return (
        <>
            <h1>{saludo}{persona}</h1>
            <h2>Mi primera Aplicación con React</h2>
            <h2>Esta cadena es un Array numérico: {numerosSeparados}</h2>
            <h2>Imprimiendo un Objecto:</h2><pre>{JSON.stringify(imprimeObjeto, null, 3)}</pre>
            <p>{subtitulo}</p>
        </>
        //Si no queremos añadir divs vacíos tenemos la opción de usar Fragment. Los podremos
        //importar o no, usando la etiqueta <Fragment> o <> respectivamente.
    );
}
//Definiendo el saludo como PropTypes para forzar a que tenga una formato específico
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo: "Zona para los subtítulos"
}

export default PrimeraApp;