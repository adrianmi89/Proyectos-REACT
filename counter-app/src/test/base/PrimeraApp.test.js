
//import {render} from "@testing-library/react";
import React from "react";
import "@testing-library/jest-dom";
import {shallow} from "enzyme";//El shallow muy parecido al render pero con algunas ventajas
import PrimeraApp from "../../PrimeraApp";

describe("Pruebas con la PrimeraApp",() => {
    /*
    test("Debe de mostrar el mensaje 'Hola, me llamo Link'",(done) => {

       const saludo = "me llamo link.";
       const {getByText} = render(<PrimeraApp persona={saludo} />);//Función del objeto wrapper

       expect (getByText(saludo)).toBeInTheDocument();

       done();
    })
    */
    test("Debe de mostrar <PrimeraApp /> correctamente",() => {
        
        const saludo = "Me llamo Link";
        console.log(saludo);
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
    })
    test("Debe de mostrar el subtítulo enviado por props",() => {
        
        const saludo = "Me llamó Link";
        const subtitulo = "Zona para los subtítulos";
        const historia = "Soy el personaje principal. Un héroe de la saga Zelda que despertó para poder proteger a la princesa.";
        
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);
        
        const textoParrafo = wrapper.find("p").text();//Extraemos de la etiqueta lo que nos interese
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subtitulo);
        //expect(wrapper).toMatchSnapshot();
    })
})