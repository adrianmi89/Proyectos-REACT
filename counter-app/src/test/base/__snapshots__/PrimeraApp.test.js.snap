// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con la PrimeraApp Debe de mostrar <PrimeraApp /> correctamente 1`] = `ShallowWrapper {}`;

exports[`Pruebas con la PrimeraApp Debe de mostrar el subt√≠tulo enviado por props 1`] = `ShallowWrapper {}`;
